pkg/services/clustering/cluster_manager.go

```diff
@@ -128,15 +142,55 @@ func (cm *ClusterManager) handleDispatcherTaskStatus(taskStatus *DispatcherTaskS
 		if taskStatus.success {
 			cm.changeAlertingState(m.CLN_ALERT_STATUS_PROCESSING)
 		}
+	case DISPATCHER_TASK_TYPE_CLEANUP:
+		if taskStatus.success {
+			cm.log.Info("Cleanup task completed")
+			cm.changeAlertingStateAndRunType(m.CLN_ALERT_STATUS_READY, m.CLN_ALERT_RUN_TYPE_NORMAL)
+		}
 	default:
 		cm.log.Error("Status received on unsupported task type "+string(taskStatus.taskType),
 			"status", taskStatus.success, "error", taskStatus.errmsg)
 	}
 
 	if !taskStatus.success {
-		cm.log.Error("Failed to dispatch task", "error", taskStatus.errmsg)
-		cm.changeAlertingState(m.CLN_ALERT_STATUS_READY)
+		cm.log.Error("Failed to dispatch/execute task", "error", taskStatus.errmsg)
+		cm.changeAlertingStateAndRunType(m.CLN_ALERT_STATUS_READY, m.CLN_ALERT_RUN_TYPE_NORMAL)
+	}
+}
+
+func (cm *ClusterManager) cleanupScheduler() bool {
+	if cm.alertingState.status != m.CLN_ALERT_STATUS_READY {
+		return false
+	}
+	tscmd := &m.GetLastDBTimeIntervalQuery{}
+	if err := bus.Dispatch(tscmd); err != nil {
+		cm.log.Error("Failed to get last heartbeat", "error", err)
+		return true
+	}
+	ts := tscmd.Result
+	checkcmd := &m.ClusteringCleanupCheckCommand{LastHeartbeat: ts}
+	if err := bus.Dispatch(checkcmd); err != nil {
+		cm.log.Error("Cleanup check failed", "error", err)
+		return true
+	}
+	if !checkcmd.Result {
+		cm.log.Info("Cleanup check : cleanup is already done")
```

Comment:
This info message should say 'Cleanup is complete'. 'Cleanup is already done' suggests that cleanup was not performed by current node but by some other node.